### PR TITLE
[HttpKernel] Move duplicated logic from Esi/Ssi to an AbstractSurrogate

### PR DESCRIPTION
**Pull request by chalasr**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

This moves the not-specific logic from the Esi/Ssi classes into an abstract class that they extend.
This class (`AbstractSurrogate`) can be extended by any class implementing a Surrogate-Capability to the Request+Response instances, as Esi/Ssi do for ESI/SSI capabilities.

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/HttpCache/Esi.php [hunk 1]
```diff
@@ -198,52 +109,7 @@ public function process(Request $request, Response $response)
...
-        } catch (\Exception $e) {
-            if ($alt) {
-                return $this->handle($cache, $alt, '', $ignoreErrors);
-            }
-
-            if (!$ignoreErrors) {
-                throw $e;
-            }
-        }
+        // remove SSI/1.0 from the Surrogate-Control header
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/HttpCache/Esi.php]: Comment looks wrong.

**Comment by fabpot** [comment]
Thank you @chalasr.

File: src/Symfony/Component/HttpKernel/HttpCache/AbstractSurrogate.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+/**
+ * Abstract class implementing Surrogate capabilities to Request and Response instances.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+abstract class AbstractSurrogate implements SurrogateInterface
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/HttpCache/AbstractSurrogate.php]: this should be either marked as `@internal` or the PR is a new feature in contrast to the PR header

Comment (chalasr) [inline src/Symfony/Component/HttpKernel/HttpCache/AbstractSurrogate.php]: @Tobion I marked it as a new feature in the header (as made in the merge commit) & added a description.

**Pull request merged**